### DISCUSSION TITLE
Vite

### DISCUSSION THREAD

**Discussion by lishaduck** [PHS-TSA/webmaster-22-23#35] [Ideas]
Hello! So, I've been messing around with the code and found some annoyances, mostly with asset handling.

1. Terrible intellisense in VSCode. JS doesn't let you type things, so I've had to resort to implicit typing via JSDoc.
2. Base handling. Getting both the dev server and the production website running is a pain.
3. Bunding/Minification. We aren't using a minifier or a bundler, so our code is slow to fetch.
4. Writing a Service Worker. I thought "Let's make this a PWA!" and immediately regretted it.
5. Modulising code. Our code was a script, and when I got rid of globalized variables, well, the ESLint output wasn't pretty.

Vite, pronounced "Veet," (it's French) is the solution. It lets us:

1. Use Typescript without dealing with TS compilation. Just pretend it's JS!
2. Well, it has base handling support, and comes with a dev server built in!
3. Uses Rollup and ESBuild to bundle code.
4. Has plugin support. Lets you interface the plugin API to get fonts and PWAs plugged in.
5. Way better module support.

However, it's not perfect. Cons:

1. Uses Node.js. Well, that's the main con. The setup means using GH Codespaces, running `npm install`, and configuring VSCode. I've done most of that for me, and it's not that hard to export it, but it still isn't ideal if you want to 

CC: @Ash-Greninja101, @InstallBipole, @NolanBallard5

**Comment by Tanza3D** [comment]
Was I accidentally mentioned here? I got a notification for this, but I have no clue what this is.
